### PR TITLE
[EOC-516] Disable getting lists from archived cohorts by getAvailableLists controller

### PR DESCRIPTION
**Pull request by alemikolo**

### CODE REVIEW

File: server/controllers/list.js [hunk 1]
```diff
@@ -1794,11 +1794,22 @@ const getAvailableLists = (req, resp) => {
...
+      .exec();
+
+    const listToSend = lists
+      .filter(
+        list => !list.cohortId || (list.cohortId && !list.cohortId.isArchived)
+      )
+      .map(({ _id, name }) => ({ _id, name }));
+
+    resp.send(listToSend);
+  } catch {
```
Comment (Ciunkos) [inline server/controllers/list.js]: Please at least log the error to ease the debugging.

File: server/controllers/list.js [hunk 2]
```diff
@@ -1794,11 +1794,22 @@ const getAvailableLists = (req, resp) => {
...
-    .catch(() => resp.sendStatus(400));
+  try {
+    const lists = await List.find(query, 'name')
+      .lean()
+      .populate('cohortId', 'isArchived')
+      .exec();
+
+    const listToSend = lists
+      .filter(
+        list => !list.cohortId || (list.cohortId && !list.cohortId.isArchived)
```
Comment (Ciunkos) [inline server/controllers/list.js]: `list.cohortId.isArchived`? Shouldn't you populate the `cohort` field instead (based on the `cohortId` field)?

File: server/controllers/list.js (hunk 1 above)
Comment (alemikolo) [reply server/controllers/list.js]: We don't console log errors. But I added a task: https://jira2.sanddev.com/browse/EOC-525,
to discuss this and create some logging system or at least console.log errors.

File: server/controllers/list.js (hunk 2 above)
Comment (alemikolo) [reply server/controllers/list.js]: we have cohortId field in list schema. when we populate cohort data it is stored in cohortId.

**Pull request merged**